docs/publishing/index.md

```diff
@@ -0,0 +1,135 @@
+
+## Do export side effects separately
+
+A [side effect](<https://en.wikipedia.org/wiki/Side_effect_(computer_science)>) in javascript is created when the script changes state outside of the scope in which it is run. This includes when code in an es module alters values available on the `window` or `document`. A side effect occurs when registering Custom Elements via `customElements.define()`. JS files with side effects should exist separately from your packages entry point. `npm init @open-wc` takes care of this for you by creating a separate `my-element.js` that manages the registration of your custom element.
+
+If you do choose to publish code with side effects as part of your entry point, it should be placed in a separate function that has to explicitly be called by the consumer (e.g. `run()`, `init()`, etc.). This allows your users to determine when these effects are triggered.
+
+## Do import 3rd party node modules with "bare" import specifiers
+
+When referring to 3rd party dependencies in your code, you cannot guarantee that implementing projects rely on the same file structure as you do. To prevent dependency specifiers from causing issues across different environments, be sure to import these dependencies into your project with "bare" import specifiers so that tooling in your users' environment can manage resolving the location of those dependencies; e.g. `import { LitElement } from 'lit-element';`
+
+## Do include file extensions in import specifiers
+
+When importing something that doesn't point directly to the entry point of a module, e.g.: `import {LitElement} from 'lit-element';`, always make sure to include a file extension: `import { ifDefined } from 'lit-html/directives/if-defined.js';`
+
+**Yes:** 👍
+
+```
+import { html } from 'lit-html';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+```
+
+**No:** 👎
+
+```
+import { html } from 'lit-html.js';
+import { ifDefined } from 'lit-html/directives/if-defined';
+```
+
+## Optional: use `type: "module"`
+
+Using `type: "module"` in your `package.json` will help CDNs and bundlers choose whether to parse your package with module parse goal. However, it will also enforce strict realities on how es modules and common JS modules must be imported into your scripting. If all of the JS in your package is es modules, and using this entry in your `package.json` does not cause issues with your scripting, then using `type: "module"` will allow those CDNs or bundlers to better leverage your package. If scripting in your package has issues running in Node@12+ due to this inclusion, excluding it will ease the local requirements on import statements in your code.
```

Comment:
```suggestion
Using `type: "module"` in your `package.json` will help CDNs and bundlers choose whether to parse your package with a module parse goal. However, it will also enforce that all the code you execute needs to be es modules. This includes all the dev tools (like bundlers, linters, ...) you use as well. CommonJs may still be used but requires a special way of importing. Therefore if all of the JS in your package is es modules, and using this entry in your `package.json` does not cause issues with your tools or scripts, then using `type: "module"` will allow those CDNs or bundlers to better leverage your package. If tools or script in your package have issues running in Node@12+ due to this inclusion, excluding it will ease the local requirements on import statements in your code.
```